pkg/kafka/partition/reader.go

```diff
@@ -114,7 +154,251 @@ func (p *Reader) run(ctx context.Context) error {
 	return nil
 }
 
-func (p *Reader) startFetchLoop(ctx context.Context) <-chan []Record {
+func (p *Reader) fetchLastCommittedOffset(ctx context.Context) int64 {
+	// We manually create a request so that we can request the offset for a single partition
+	// only, which is more performant than requesting the offsets for all partitions.
+	req := kmsg.NewPtrOffsetFetchRequest()
+	req.Topics = []kmsg.OffsetFetchRequestTopic{{Topic: p.kafkaCfg.Topic, Partitions: []int32{p.partitionID}}}
+	req.Group = p.consumerGroup
+
+	resps := p.client.RequestSharded(ctx, req)
+
+	// Since we issued a request for only 1 partition, we expect exactly 1 response.
+	if expected := 1; len(resps) != expected {
+		level.Error(p.logger).Log("msg", "unexpected number of responses", "expected", 1, "actual", len(resps))
```

Comment:
Perhaps this was the intention instead?

```suggestion
		level.Error(p.logger).Log("msg", "unexpected number of responses", "expected", expected, "actual", len(resps))
```

Reply:
I think so.
I toyed with removing all the expected,actual variables but I think I'm just going to use this suggestion and leave the error handling as it is. IMO it's quite verbose (I borrowed this from Mimir) but, on the other hand, it is clear from the error message what happened so I don't think it's worth changing.